File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.4. PAO202650_RESTFUL_DOTNET_SINBDD_NRC30742/04. CLICON/CONUNI_SOAP_JAVA_CLICON_G05/src/main/java/ec/edu/monster/controlador/UtilidadesConsola.java
package ec.edu.monster.controlador;

/**
 * Utilidades para la consola
 * @author ACER NITRO V15
 */
public class UtilidadesConsola {
    
    /**
     * Limpia la pantalla de la consola
     */
    public static void limpiarPantalla() {
        try {
            // Para Windows
            if (System.getProperty("os.name").toLowerCase().contains("windows")) {
                new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
            } else {
                // Para Unix/Linux/Mac
                System.out.print("\033[H\033[2J");
                System.out.flush();
            }
        } catch (Exception e) {
            // Si falla, simplemente imprime muchas líneas en blanco
            for (int i = 0; i < 50; i++) {
                System.out.println();
            }
        }
    }
    
    /**
     * Pausa la ejecución por un tiempo determinado
     */
    public static void pausar(int segundos) {
        try {
            Thread.sleep(segundos * 1000);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
    
    /**
     * Muestra una línea separadora
     */
    public static void mostrarSeparador() {
        System.out.println("=".repeat(60));
    }
    
    /**
     * Muestra una línea separadora más pequeña
     */
    public static void mostrarSeparadorPequeño() {
        System.out.println("-".repeat(40));
    }
    
    /**
     * Centra un texto en la consola
     */
    public static void centrarTexto(String texto) {
        int ancho = 60;
        int espacios = (ancho - texto.length()) / 2;
        System.out.println(" ".repeat(Math.max(0, espacios)) + texto);
    }
}

File: TI1/TI1.2 SOAP_DOTNET_SINBDD_GR09/02.CLICON/CONUNI_SOAP_DOTNET_CLICON_G05/CONUNI_SOAP_DOTNET_CLICON_G09/ec/edu/monster/controlador/UtilidadesConsola.cs
using System;

namespace ec.edu.monster.controlador
{
    /// <summary>
    /// Utilidades para la consola
    /// Replicado desde Java
    /// </summary>
    public static class UtilidadesConsola
    {
        /// <summary>
        /// Limpia la pantalla de la consola
        /// </summary>
        public static void LimpiarPantalla()
        {
            try
            {
                // Console.Clear() funciona correctamente en Windows, Linux y Mac
                Console.Clear();
            }
            catch (Exception)
            {
                // Si falla, simplemente imprime muchas líneas en blanco
                for (int i = 0; i < 50; i++)
                {
                    Console.WriteLine();
                }
            }
        }
        
        /// <summary>
        /// Pausa la ejecución por un tiempo determinado
        /// </summary>
        public static void Pausar(int segundos)
        {
            try
            {
                System.Threading.Thread.Sleep(segundos * 1000);
            }
            catch (System.Threading.ThreadInterruptedException)
            {
                System.Threading.Thread.CurrentThread.Interrupt();
            }
        }
        
        /// <summary>
        /// Muestra una línea separadora
        /// </summary>
        public static void MostrarSeparador()
        {
            Console.WriteLine(new string('=', 60));
        }
        
        /// <summary>
        /// Muestra una línea separadora más pequeña
        /// </summary>
        public static void MostrarSeparadorPequeño()
        {
            Console.WriteLine(new string('-', 40));
        }
        
        /// <summary>
        /// Centra un texto en la consola
        /// </summary>
        public static void CentrarTexto(string texto)
        {
            int ancho = 60;
            int espacios = (ancho - texto.Length) / 2;
            Console.WriteLine(new string(' ', Math.Max(0, espacios)) + texto);
        }
    }
}


